File: elixir-ms/lib/domain/usecase/gethello/get_hello_use_case.ex
defmodule ElixirMs.UseCase.GetHelloUseCase do
  @moduledoc false

  @hello_repository Application.compile_env(:elixir_ms, :hello_repository)

  def hello_http(latency) do
    @hello_repository.hello_http(latency)
  end

  def hello_https(latency) do
    @hello_repository.hello_https(latency)
  end

  def hello_connection_pool_http1(latency) do
    @hello_repository.hello_connection_pool_http1(latency)
  end

  def hello_connection_pool_http2(latency) do
    @hello_repository.hello_connection_pool_http2(latency)
  end
end

File: java-reactive-ms/domain/usecase/src/main/java/com/maocq/usecase/gethello/GetHelloUseCase.java
package com.maocq.usecase.gethello;

import com.maocq.model.hello.gateways.HelloRepository;
import lombok.RequiredArgsConstructor;
import reactor.core.publisher.Mono;

@RequiredArgsConstructor
public class GetHelloUseCase {

    private final HelloRepository helloRepository;

    public Mono<String> helloHttp(int latency) {
        return helloRepository.helloHttp(latency);
    }

    public Mono<String> helloHttps(int latency) {
        return helloRepository.helloHttps(latency);
    }

    public Mono<String> helloConnectionPoolHttp1(int latency) {
        return helloRepository.helloConnectionPoolHttp1(latency);
    }

    public Mono<String> helloConnectionPoolHttp2(int latency) {
        return helloRepository.helloConnectionPoolHttp2(latency);
    }
}
